.ci/setenvconfig

```diff
@@ -0,0 +1,303 @@
+#!/usr/bin/env bash
+
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+# Script to setup environment variables and deployer config to run CI tasks.
+#
+# Usage: setenvconfig ENV
```

Comment:
I fear that it will be difficult to keep this doc up to date.
I wonder if it would be better to generate this document.

```sh
# e2e/aks)

$BUILD_TAG

$VAULT_ADDR
$VAULT_ROLE_ID
$VAULT_SECRET_ID

# e2e/custom-operator-image)

$GCLOUD_PROJECT

$BUILD_NUMBER
$JKS_PARAM_OPERATOR_IMAGE
$JKS_PARAM_K8S_VERSION

$VAULT_ADDR
$VAULT_ROLE_ID
$VAULT_SECRET_ID

# e2e/gke-k8s-versions)

clusterVersion=$2
clusterName=$3

$GCLOUD_PROJECT

$JKS_PARAM_OPERATOR_IMAGE

$VAULT_ADDR
$VAULT_ROLE_ID
$VAULT_SECRET_ID

# e2e/kind-k8s-versions)

kindNodeImage=$2

$GCLOUD_PROJECT

$JKS_PARAM_OPERATOR_IMAGE

# e2e/master)

$GCLOUD_PROJECT

$BUILD_NUMBER

$VAULT_ADDR
$VAULT_ROLE_ID
$VAULT_SECRET_ID

# e2e/stack-versions)

clusterName=$2
stackVersion=$3

$GCLOUD_PROJECT

$JKS_PARAM_OPERATOR_IMAGE

$VAULT_ADDR
$VAULT_ROLE_ID
$VAULT_SECRET_ID

# pr)

$GCLOUD_PROJECT

$BUILD_NUMBER

$VAULT_ADDR
$VAULT_ROLE_ID
$VAULT_SECRET_ID

# build/release)

$WORKSPACE
$TAG_NAME

# build/nightly)

$WORKSPACE

# cleanup)

$GCLOUD_PROJECT

$JKS_PARAM_GKE_CLUSTER

$VAULT_ADDR
$VAULT_ROLE_ID
$VAULT_SECRET_ID
```

Reply:
Ah, I meant just common ones. I now think I'm fine with just saying that params depend on the job we are doing the setup for, because the only common ones are vault ones and these may not be required depending on what we do with https://github.com/elastic/cloud-on-k8s/pull/2502#discussion_r376887858.